src/execution_plan/ops/op_update.c

```diff
@@ -71,6 +72,34 @@ void _OpUpdate_QueueUpdate(OpUpdate *op, EntityProperty *dest_entity_prop, SIVal
     op->entities_to_update_count++;
 }
 
+/* If a new property was introduced, add it to the schema associated
+ * with its label. */
+void _UpdateSchema(const OpUpdate *op, EntityUpdateEvalCtx *expression, EntityID id) {
```

Comment:
I feel like this is going to be changed, in case we feel like the introduction of triemap per updated node label, which keeps track after which properties were  we're touched/mentioned is sufficient.